eth/input.go

```diff
@@ -0,0 +1,51 @@
+package eth
+
+import (
+	"github.com/INFURA/go-ethlibs/rlp"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+type Input Data
+
+func NewInput(value string) (*Input, error) {
+	if !strings.HasPrefix(value, "0x") {
+		return nil, errors.Errorf("invalid input: %s", value)
+	}
+
+	a := Input(value)
+	return &a, nil
+}
+
+func MustInput(value string) *Input {
+	a, err := NewInput(value)
+	if err != nil {
+		panic(err)
+	}
+
+	return a
+}
+
+func (i Input) String() string {
+	return string(i)
+}
+
+func (i Input) Bytes() []byte {
+	return Data20(i).Bytes()
```

Comment:
```suggestion
	return Data(i).Bytes()
```

Reply:
afaik there's no functional difference but imo using `Data` is more readable here since `Input` could be > than 20 bytes.